File: Stacks/10/validParen.js
var isValid = function (s) {
  if (s === '') return true;

  const stack = []
  for (let ch of s) {
    switch(ch) {
      case '(': stack.push(ch); break;
      case '{': stack.push(ch);break;
      case '[': stack.push(ch);break;
      case ')': 
        if (stack.length){
          popped = stack.pop();
          if (popped === '(') break;
          else return false;
        } else return false;
      case '}': 
        if (stack.length){
          popped = stack.pop();
          if (popped === '{') break;
          else return false;
        } else return false;
      case ']':
        if (stack.length){
          popped = stack.pop();
          if (popped === '[') break;
          else return false;
        } else return false;
      default:
        return false;
    }
  }
  if (stack.length) return false;
  else return true;
};


const ch = '';
console.log(isValid(ch));

File: Stacks/10/validParen.py

def isValid(s):
    if s == "":
        return True
    stack = []
    for ch in s:
        if ch == '(' or ch == '{' or ch == '[':
            stack.append(ch)
        else:
            if len(stack) > 0:
                popped = stack.pop()
                if ch == '}':
                    if popped == '{':
                        continue
                    else:
                        return False
                elif ch == ')':
                    if popped == '(':
                        continue
                    else:
                        return False
                elif ch == ']': 
                    if popped == '[':
                        continue
                    else:
                        return False
                else:
                    return False
            else:
                return False
    if len(stack):
        return False
    else:
        return True # Made mistake of returning True as soon as for loop is done

ch = '([])'
print(isValid(ch))
